src/CMakeLists.txt

```diff
@@ -218,7 +221,8 @@ function(CUDA_CONVERT_FLAGS EXISTING_TARGET)
 endfunction()
 
 if (USE_CUDA)
-  if (${CMAKE_VERSION} VERSION_LESS 3.13)
+  # issue seems to be present even with CMake 3.14.5 in some environments
+  #if (${CMAKE_VERSION} VERSION_LESS 3.13)
```

Comment:
According to https://gitlab.kitware.com/cmake/cmake/issues/18897 there was a second issue originating from the linker flags, which has been resolved in CMake 3.15.x.
Which version did you use for testing? Because if it's newer than 3.16 and we still observe it we should report it.

Reply:
v3.14.5

I tested with CUDA 10.1.x on Power9 using the GNU toolchain.

Reply:
I can upgrade the condition to 3.16...

Reply:
I am compiling v3.16.0-rc4 and double-check.

Reply:
Yes, please do that. I'd rather have another error again once somebody tries with a recent enough version of CMake such that we can report an issue upstream.